odoo/tools/misc.py

```diff
@@ -1254,13 +1254,55 @@ def __gt__(self, other): return self.val < other.val
     def __le__(self, other): return self.val >= other.val
     def __lt__(self, other): return self.val > other.val
 
+# there is contextlib.suppress since py3
 @contextmanager
 def ignore(*exc):
     try:
         yield
     except exc:
         pass
 
+class reraise_x_as(ContextDecorator):
+    """
+    Hide some exceptions behind a fake alternative error. Can be used as
+    a function decorator or as a context manager.
+
+    .. code-block:
+
+        @route('/super/secret/route', auth='public')
+        @reraise_x_as(AccessError, as_=NotFound())
+        def super_secret_route(self):
+            if not request.session.uid:
+                raise AccessError("Route hidden to non logged-in users")
+            ...
+
+        def some_util():
+            ...
+            with reraise_x_as(ValueError, as_=UserError("Invalid argument")):
+                ...
+            ...
+
+    :param exceptions: the exception classes to except.
```

Comment:
As a verb, "to except" means "to exclude". This should be "catch". Though "replace" or "substitute" would be more precise contextually.